### PR TITLE
tests: Added skip messages to tests

### PR DESCRIPTION
**Pull request by Whenning42**
Went through all the tests in layer_validation_tests and added skip
messages to all tests that conditionally skip their checks.
Additionally, added the prefix "TEST_SKIPPED" to all the skip messages
so that they can easily be parsed.

Also added a script that looks parses the output of all the tests and
prints a short summary of which tests were skipped on every device
profile and which tests print unexpected output.

### CODE REVIEW

File: tests/layer_validation_tests.cpp [hunk 1]
```diff
@@ -2276,6 +2278,7 @@ TEST_F(VkLayerTest, InvalidMemoryMapping) {
     pass = m_device->phy().set_memory_type(mem_reqs.memoryTypeBits, &alloc_info, VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT,
                                            VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
     if (!pass) {
+        printf("             TEST_SKIPPED: Failed to set memory type.\n");
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: This is a sub portion of a test skipped... there are other example w/ more descriptive text about this.

File: tests/layer_validation_tests.cpp [hunk 2]
```diff
@@ -2303,7 +2306,7 @@ TEST_F(VkLayerTest, MapMemWithoutHostVisibleBit) {
     pass = m_device->phy().set_memory_type(0xFFFFFFFF, &mem_alloc, 0, VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT);
     if (!pass) {  // If we can't find any unmappable memory this test doesn't
                   // make sense
-        printf("             No unmappable memory types found, skipping test\n");
+        printf("             TEST_SKIPPED: No unmappable memory types found, skipping test\n");
```
Comment (jzulauf-lunarg) [inline tests/layer_validation_tests.cpp]: well... not this one, clearly.

File: tests/layer_validation_tests.cpp [hunk 3]
```diff
@@ -55,24 +55,26 @@
...
-    BsoFailStencilWriteMask,
-    BsoFailStencilReference,
-    BsoFailCmdClearAttachments,
-    BsoFailIndexBuffer,
-    BsoFailIndexBufferBadSize,
-    BsoFailIndexBufferBadOffset,
-    BsoFailIndexBufferBadMapSize,
-    BsoFailIndexBufferBadMapOffset
-};
+const char *skip_prefix = "             TEST SKIPPED:"
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: Like having this const. Update var name to kSkipPrefix.
For const names prefix w/ "k" and use camel-case per [style guide](https://google.github.io/styleguide/cppguide.html#Constant_Names)

File: tests/layer_validation_tests.cpp [hunk 4]
```diff
@@ -55,24 +55,26 @@
...
+
+    enum BsoFailSelect {
```
Comment (tobine) [inline tests/layer_validation_tests.cpp]: Did you intend to indent this enum? Seems like it should be shifted back over as it was.

**Comment by tobine** [comment]
@jzulauf-lunarg @Whenning42 and I discussed the partial skip case and my feeling was that if every profile is partially skipping it's still an indication that a portion of the test did not run at all. The primary use of the tag will be to identify cases when a test is skipped on all profiles so the equal treatment of partial or complete skips is acceptable to me.

**Pull request merged**